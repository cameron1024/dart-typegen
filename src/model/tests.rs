use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
