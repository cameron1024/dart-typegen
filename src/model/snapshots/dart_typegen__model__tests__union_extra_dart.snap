---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    meta: None,
    defaults: None,
    enums: [],
    classes: [],
    unions: [
        Union {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Animal",
                span: SourceSpan {
                    offset: SourceOffset(
                        47,
                    ),
                    length: 8,
                },
            },
            sealed: None,
            json_discriminant: Some(
                SpannedScalar {
                    value: "__type",
                    span: SourceSpan {
                        offset: SourceOffset(
                            78,
                        ),
                        length: 8,
                    },
                },
            ),
            json_discriminant_value_case: None,
            annotations: None,
            builder_annotations: None,
            docs: None,
            classes: [
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Dog",
                        span: SourceSpan {
                            offset: SourceOffset(
                                96,
                            ),
                            length: 5,
                        },
                    },
                    allow_non_pascal_case: None,
                    fields: [],
                    docs: None,
                    json_key_case: None,
                    json_discriminant_value: None,
                    annotations: None,
                    builder_annotations: None,
                    extra_dart: None,
                    builder_extra_dart: None,
                },
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Cat",
                        span: SourceSpan {
                            offset: SourceOffset(
                                110,
                            ),
                            length: 5,
                        },
                    },
                    allow_non_pascal_case: None,
                    fields: [],
                    docs: None,
                    json_key_case: None,
                    json_discriminant_value: None,
                    annotations: None,
                    builder_annotations: None,
                    extra_dart: Some(
                        SpannedScalar {
                            value: "\n      void printCat() => print(this);\n    ",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    133,
                                ),
                                length: 48,
                            },
                        },
                    ),
                    builder_extra_dart: None,
                },
            ],
            extra_dart: [
                SpannedScalar {
                    value: "\n    void printAnimal() => print(this);\n  ",
                    span: SourceSpan {
                        offset: SourceOffset(
                            200,
                        ),
                        length: 47,
                    },
                },
            ],
        },
    ],
}
